### ISSUE TITLE
How is it that the Prototypical Nets get high accuracy without training?

### ISSUE THREAD

**Issue by cinjon** [tristandeleu/pytorch-meta#41]
If I go into the prototypical nets example and turn off training by using torch.no_grad befoe the tqdm loop and then commenting out the `model.zero_grad`, `loss.backward`, and `optimizer.step`, I get accuracy in the high 70s. 

How is this happening? I would expect that accuracy would be 20%. I changed the model by removing BatchNorm and ReLU and it gets high 60s in this case. I thought it *could* be something to do with the convolutional prior, but then I changed that to just a Linear function and it got accuracy in the high 60s as well.

Is it a bug in the implementation?

**Comment by carmete** [comment]
Even a randomly initialised network will propagate some structure from the input data to the output layer. Thus, if the classification task is not to hard, samples from the same class will still lie close enough in the space in which prototypes are calculated.

Which data set are you using? For Omniglot your results seem reasonable.

**Comment by cinjon** [comment]
This is omniglot, and yes, sorrry, should have updated this post that it's actually reasonable.
